src/join_mesh.py

```diff
@@ -1,140 +1,254 @@
 #!/usr/bin/env python3
+"""
+Joins meshes partitioned by ASTE mesh partitioner.
+
+There is two possible way of merging:
+- Partition-wise (does not recovery original mesh order and lack of discarded cells)
+- Recovery merge (maintain original mesh order and adds discarded cells)
```

Comment:
When we partition the mesh, the cells on the borders are discarded and saved into a recovery file. 
Since both preCICE and ASTE provide partitioned mesh in `vtu` format only `vtu` files are searched and merged.
In terms of joining or writing `vtk` and `vtu` format, there is no difference, just writer for them is different.

Reply:
Could you add this as explanation above? If this is something typical for vtk, maybe there is also a section in their documentation to refer to.